### DISCUSSION TITLE
messages parameter modified by the completion execution

### DISCUSSION THREAD

**Discussion by miguelwon** [BerriAI/litellm#4489] [General]
The message list parameter in `completion` is not being copied, which results in list modification when `completion` is executed. 
Is this the desired behaviour?

```
from litellm import completion
import os
os.environ["ANTHROPIC_API_KEY"] = ""

messages = [
    {'role':'system', 'content':"Your are a helpfull AI assistant"},
    {'role':'user', 'content':"Hello!"}
]

response = completion(
    model="claude-3-5-sonnet-20240620",
    messages=messages
)

print(messages)
[{'role': 'user', 'content': 'Hello!'}]
```

**Comment by ghost** [comment]
@miguelwon i believe this is system being popped and moved to the system param for anthropic. [Code](https://github.com/BerriAI/litellm/blob/c31156e5e690f18a4941abcccfe8310b56efab4b/litellm/llms/anthropic.py#L547)
We can probably just deepcopy this before modifying, to prevent any issues.

**Comment by ghost** [comment]
Hey @miguelwon fixed here - https://github.com/BerriAI/litellm/commit/be3c98bc162bdeea8aaae2a889058203f28b40e8

Can we do a 10 min call this week? Would love to learn how you're using litellm 

https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat
